angular-intro-project/src/app/courses.reducer.ts

```diff
@@ -0,0 +1,17 @@
+import { updateCourses, loadMoreCourses } from './courses.actions';
+import { createReducer, on } from '@ngrx/store';
+
+export const initialState = '';
+
+const _coursesReducer = createReducer(initialState,
+  on(updateCourses, (state, { courseItems }) => {
+    return courseItems;
+  }),
+  on(loadMoreCourses, (state, { courseItems }) => {
+    return state.concat(courseItems);
```

Comment:
We should return new state and not mutate the previous one. This is one of the main concepts of redux.